### ISSUE TITLE
Option to mark warnings as errors

### ISSUE THREAD

**Issue by vanniktech** [google/error-prone#424]
I'd love an option to mark error prone warnings as errors so that even warnings won't get ignored.

While it is possible to do the following:

``` groovy
tasks.withType(JavaCompile) {
  options.compilerArgs += [ '-Werror' ]
}
```

it does not really suffice since all warnings are then marked as errors. Another option would be for every error prone rule to increase the severity though this will get impossible to maintain.

Maybe it could be something as simple as `-Xep:WarningsAsErrors`.

**Comment by cushon** [comment]
This sounds like a good idea to me. We've also received requests for a flag to enable all checks as warnings (even if they're errors by default) in order to collect the full list of diagnostics for a codebase. I imagine the implementation of the two flags would be very similar.

**Comment by vanniktech** [comment]
@cushon wasn't this implemented the other way around meaning that you can downgrade errors to warnings but not the other way around?

**Comment by emartynov** [comment]
Is it implemented?

**Comment by emartynov** [comment]
Thanks! Would be nice to re-open it :)

**Comment by jbduncan** [comment]
Indeed! :)

But alas I'm not a Googler, so I don't have permissions to reopen this issue.

@cushon WDYT?

**Comment by msridhar** [comment]
Agreed this should be re-opened, but I'm curious about the downsides of `-Werror`; it seems to be good enough for us.  Which built-in javac warnings do folks find to be to noisy?

**Comment by emartynov** [comment]
We have one warning about using deprecated API and it will break the build now. I don't know if I would want to mix deprecation and error-prone warnings. But probably, any warning should be caught and fixed as soon as possible.

**Comment by msridhar** [comment]
> We have one warning about using deprecated API and it will break the build now. I don't know if I would want to mix deprecation and error-prone warnings. But probably, any warning should be caught and fixed as soon as possible.

Can you use `-Xlint:-deprecation` to disable the deprecation lint as a workaround?
